complete main functionality of a trainer

mmarty12 opened the pull request.

AlinaDubchak committed:
added option: color correction of pressed keys

AlinaDubchak committed:
added text processing by words and sentences

AlinaDubchak committed:
corrected the comments to the previous commit

mmarty12 committed:
added the display of the typed text

+ the typed text changes colour to a blank one. Functionality was implemented for both variants (string of letters and text).

AlinaDubchak committed:
added check for the correctness of pressed keys and their illumination in case of writing an incorrect character. Also added an inscription to the input

mmarty12 committed:
added implementation of typing statistic

count and display of symbols/min, words/min, error percentage.

AlinaDubchak committed:
added tests for function press

AlinaDubchak committed:
added test for viewUpdate function + change some argumets for function

AlinaDubchak committed:
added tests for easy lvl

AlinaDubchak committed:
added CI/CD

AlinaDubchak reviewed (commented):
left several comments

AlinaDubchak commented on script.js:
```diff
@@ -0,0 +1,211 @@
+const input = document.querySelector('input');
+const letters = Array.from(document.querySelectorAll('[data-letters]'));
+const specs = Array.from(document.querySelectorAll('[data-spec]'));
+const textExample = document.querySelector('#textExample');
+const symbolsPerMinute = document.querySelector('#symbolsPerMinute');
+const wordsPerMinute = document.querySelector('#wordsPerMinute');
+const errorPercent = document.querySelector('#errorPercent');
+
+const text = `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in...`;
+const textLength = text.length;
+
+const party = createParty(text);
```
prefer use object for this constants

AlinaDubchak commented on script.js:
```diff
@@ -0,0 +1,211 @@
+}
+
+function createParty(text) {
+  const party = {
+    text,
+    strings: [],
+    maxStringLength: 70,
+    maxShowStrings: 5,
+    currentStringIndex: 0,
+    currentPressedIndex: 0,
+    errors: [],
+    started: false,
+    startTimer: 0,
+    timerCounter: 0,
+    symbolCounter: 0,
+    wordCounter: 0,
+    errorCounter: 0,
+    statisticFlag: false,
+  };
+  party.text = party.text.replace(/\n/g, '\n ');
```
it is prefer to export to a global object

AlinaDubchak commented on script.js:
```diff
@@ -0,0 +1,211 @@
+}
+
+function viewUpdate(party,textExample) {
+  const string = party.strings[party.currentStringIndex];
```
const string = party.strings[party.currentStringIndex];
const shownString = party.strings.slice(...)
const firstLine = document.createElement('div');
let div = document.createElement('div');
const doneLine = document.createElement('span');

prefer use a object for this variables.

AlinaDubchak commented on script.js:
```diff
@@ -0,0 +1,211 @@
+      input.removeEventListener('keydown', keyDownHandler);
+      input.removeEventListener('keyup', keyUpHandler);
+    }
+  }
```
also better to rewrite the function `viewUpdate()` in a more readable form

AlinaDubchak commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,18 @@
+
+on:
+  push:
+    branches: [ trainer_mainpage_back ]
```
change brang to `main`

AlinaDubchak commented on scriptEasy.js:
```diff
@@ -0,0 +1,201 @@
+  if (party.currentPressedIndex === textLength) {
+    statisticCount();
+  }
+}
```
make the same changes as in the file `script.js`

mmarty12 committed:
exported party as a global object

mmarty12 committed:
refactored viewUpdate function

mmarty12 committed:
refactored the "string"-trainer variant

Made the same improvements as for the script.js:
- made the party object global
- refactored viewUpdate function;

mmarty12 committed:
packed constants in one settings object

Did so for both variants

mmarty12 committed:
test configure corrections
